src/cargo/core/compiler/build_config.rs

```diff
@@ -1,11 +1,12 @@
 use crate::core::compiler::CompileKind;
 use crate::util::interning::InternedString;
 use crate::util::{CargoResult, Config, RustfixDiagnosticServer};
-use anyhow::bail;
+use anyhow::{bail, Context as _};
 use cargo_util::ProcessBuilder;
 use serde::ser;
 use std::cell::RefCell;
 use std::path::PathBuf;
+use std::thread::available_parallelism;
```

Comment:
What makes this safe to do now compared to the concerns in https://github.com/rust-lang/cargo/pull/10737?

Reply:
It looks like rust-lang/rust#97925 fixed the issue with `available_parallelism` and cgroups v1 on Linux

Reply:
I've updated the PR message to call that out.